test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -698,7 +703,7 @@ TEST(LoopNest, SplitWithTailWithLoopOptions) {
   ASSERT_GT(loops.size(), 0);
   l.setGPUBlockIndex(loops[0], LoopOptions::IDX_Y);
   l.splitWithTail(loops[0], 4, &outer, &inner, &tail);
-  ASSERT_NE(outer, nullptr);
+  ASSERT_EQ(loops[0], outer);
```

Comment:
Please also add a test checking that loops from the body remain valid after the split as well (see a comment to the second PR in the stack for an example).

Reply:
Will add this test once `IRMutator` and `Substitute` APIs are fixed.